community/modules/compute/schedmd-slurm-gcp-v6-nodeset/source_image_logic.tf

```diff
@@ -1,77 +0,0 @@
-  #
-  # https://github.com/terraform-google-modules/terraform-google-vm/blob/735bd415fc5f034d46aa0de7922e8fada2327c0c/modules/instance_template/main.tf#L28
-  # https://cloud.google.com/apis/design/resource_names#relative_resource_name
-  source_image_project_normalized = (can(var.instance_image.family) ?
-    "projects/${data.google_compute_image.slurm.project}/global/images/family" :
-    "projects/${data.google_compute_image.slurm.project}/global/images"
-  )
-  source_image_family = can(var.instance_image.family) ? data.google_compute_image.slurm.family : ""
-  source_image        = can(var.instance_image.name) ? data.google_compute_image.slurm.name : ""
-}
-
-data "google_compute_image" "slurm" {
-  family  = try(var.instance_image.family, null)
-  name    = try(var.instance_image.name, null)
-  project = var.instance_image.project
-
-  lifecycle {
-    precondition {
-      condition     = length(regexall("^projects/.+?/global/images/family$", var.instance_image.project)) == 0
-      error_message = "The \"project\" field in var.instance_image no longer supports a long-form ending in \"family\". Specify only the project ID."
-    }
-
-    postcondition {
-      condition     = var.instance_image_custom || contains(keys(local.known_project_families), self.project)
-      error_message = <<-EOD
-      Images in project ${self.project} are not published by SchedMD. Images must be created by compatible releases of the Terraform and Packer modules following the guidance at https://goo.gle/hpc-slurm-images. Set var.instance_image_custom to true to silence this error and acknowledge that you are using a compatible image.
-      EOD
-    }
-    postcondition {
-      condition     = !contains(keys(local.known_project_families), self.project) || try(contains(local.known_project_families[self.project], self.family), false)
-      error_message = <<-EOD
-      Image family ${self.family} published by SchedMD in project ${self.project} is not compatible with this release of the Terraform Slurm modules. Select from known compatible releases:
-      ${join("\n", [for p in try(local.known_project_families[self.project], []) : "\t\"${p}\""])}
-      EOD
-    }
-    postcondition {
-      condition     = var.disk_size_gb >= self.disk_size_gb
-      error_message = "'disk_size_gb: ${var.disk_size_gb}' is smaller than the image size (${self.disk_size_gb}GB), please increase the blueprint disk size"
-    }
-    postcondition {
-      # Condition needs to check the suffix of the license, as prefix contains an API version which can change.
-      # Example license value: https://www.googleapis.com/compute/v1/projects/cloud-hpc-image-public/global/licenses/hpc-vm-image-feature-disable-auto-updates
-      condition     = var.allow_automatic_updates || anytrue([for license in self.licenses : endswith(license, "/projects/cloud-hpc-image-public/global/licenses/hpc-vm-image-feature-disable-auto-updates")])
```

Comment:
Just beware that removing the license check opens us up to likely more support requests - we should probably document that this feature is only available on the HPC image and Slurm Rocky Linux 8 image.